File: wiz-cross-check-java/src/main/java/com/example/wizcrosscheckjava/domain/crosscheck/SettlementCrossCheckRepository.java
package com.example.wizcrosscheckjava.domain.crosscheck;

import java.time.LocalDate;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface SettlementCrossCheckRepository extends JpaRepository<SettlementCrossCheck, Long> {

	Optional<SettlementCrossCheck> findByTargetDate(LocalDate date);
}

File: wiz-cross-check/src/main/kotlin/com/example/wizcrosscheck/domain/crosscheck/SettlementCrossCheckRepository.kt
package com.example.batchsample.domain.crosscheck

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.time.LocalDate

@Repository
interface SettlementCrossCheckRepository : JpaRepository<SettlementCrossCheck, Long> {

    fun findByTargetDate(date: LocalDate): SettlementCrossCheck?
}